Source/core/SystemInfo.cpp

```diff
@@ -195,14 +195,17 @@ namespace Core {
         sysinfo(&info);
 
         m_uptime = info.uptime;
-        m_totalram = info.totalram;
         m_pageSize = static_cast<uint32_t>(sysconf(_SC_PAGESIZE));
-        m_freeram = info.freeram;
-        m_totalswap = info.totalswap;
-        m_freeswap = info.freeswap;
         m_cpuloadavg[0]=info.loads[0];
         m_cpuloadavg[1]=info.loads[1];
         m_cpuloadavg[2]=info.loads[2];
+
+        MemorySnapshot snapshot = TakeMemorySnapshot();
```

Comment:
Hi Santhosh, the idea of the memory measurement here is that it is fast but less precise. The TakeMemorySnapshot was indeed added in our code base to be more precise but comes at a cost as it is more expensive (meaning requires more processing power).
Therefore the TakeMemorySnapshot is only used in our code to when the precision is really needed (e.g. ResourceMonitor)
So I propose to leave this like it is as here we prefer speed over precision. Do you agree?

Reply:
Yes, as long as we dont use the free mem from sysinfo for any critical memory management. So, should i close this pull then?

Reply:
Yes let's do that